File: javascript/writeFile.js
#!/usr/bin/env node

/**
 * Usage: ./writeFile.js [filepath]
 */

const fs = require("fs");

const params = process.argv.slice(2);

if (params.length > 0) {
    writeFile(params[0])
} else {
    console.error("Missing file argument.");
}

function writeFile(filepath) {
    console.log("Writing file to: " + filepath);
    // TODO: implementation
    const content = "Hello World!";
    fs.writeFile(filepath, content, (err) => {
        if (err) {
            console.error("Error writing to file: " + err);
        }
    });
}

File: python/write_file.py
#!/usr/bin/env python

# Usage: ./write_file.py [filepath]

def write_file(filepath):
    # TODO: implementation
    print("Writing file to: " + filepath)
    with open(filepath, "w") as file_content:
        file_content.write("Hello World!")

if __name__ == "__main__":
    import sys
    if len(sys.argv) > 1:
        write_file(sys.argv[1])
    else:
        print("Missing file argument.")
